### PR TITLE
helm: use toYaml for discovery nodeAffinity

### PR DESCRIPTION
**Pull request by hhk7734**
**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by travisn — CHANGES_REQUESTED** [review]
Thanks for the contribution. I see this will now have the same format that the CSI settings are already using, so it makes sense. 

Could you just squash to a single commit? Or else the second commit needs a description message. A single commit is fine since it's the same description.

**Comment by hhk7734** [comment]
@travisn 

I squashed it into a single commit :)

**Review by travisn — APPROVED** [review]

File: deploy/charts/rook-ceph/templates/deployment.yaml [hunk 1]
```diff
@@ -68,7 +68,7 @@ spec:
 {{- end }}
 {{- if .Values.discover.nodeAffinity }}
         - name: DISCOVER_AGENT_NODE_AFFINITY
-          value: {{ .Values.discover.nodeAffinity }}
+          value: {{ toYaml .Values.discover.nodeAffinity | quote }}
```
Comment (parth-gr) [inline deploy/charts/rook-ceph/templates/deployment.yaml:71]: why `quote` @hhk7734 ?

Comment (hhk7734) [reply deploy/charts/rook-ceph/templates/deployment.yaml:71]: Because it is an env vars, an error occurs if there is no `quote` when using dict type yaml.

Comment (hhk7734) [reply deploy/charts/rook-ceph/templates/deployment.yaml:71]: without `quote`

```yaml
        - name: DISCOVER_AGENT_NODE_AFFINITY
          value: |
            {{- toYaml .Values.discover.nodeAffinity | nindent 12 }}
```

Comment (parth-gr) [reply deploy/charts/rook-ceph/templates/deployment.yaml:71]: yes without quote make more sense to me, not sure how it is `nindent` if just specify the `quote` keyword, do you have any document that explains it

Comment (hhk7734) [reply deploy/charts/rook-ceph/templates/deployment.yaml:71]: https://helm.sh/docs/chart_template_guide/functions_and_pipelines/

Using `| quote` in `env[].value` is very common, so I cannot get the reason to replace to `nindent`.

Even you can see using `quote` above this line.

Comment (parth-gr) [reply deploy/charts/rook-ceph/templates/deployment.yaml:71]: thanks for the info!

**Pull request merged**